src/os/bluestore/BlueFS.cc

```diff
@@ -3386,17 +3389,18 @@ int BlueFS::truncate(FileWriter *h, uint64_t offset)/*_WF_L*/
   }
   ceph_assert(h->file->fnode.size >= offset);
   _flush_bdev(h);
+
+  std::lock_guard ll(log.lock);
```

Comment:
this rather worth a different commit as it irrelevant to volume selector check and shouldn't be lost for example in case of that check removal.

Reply:
I added comment to _check_vselector_LNF:
```
  // Checking vselector is under log, nodes and file(s) locks,
  // so any modification of vselector must be under at least one of those locks.
```